ludwig/train.py

```diff
@@ -396,6 +400,68 @@ def full_train(
     )
 
 
+def kfold_cross_validate(
+                model_definition,
+                model_definition_file=None,
+                data_csv=None,
+                data_train_csv=None,
+                data_validation_csv=None,
+                output_directory='results',
+                k_fold=None,
+                **kwargs
+):
+    logger = logging.getLogger('ludwig')
+    logger.info('starting {:d}-fold cross validation'.format(k_fold))
+
+    # create output_directory if not available
+    if not os.path.isdir(output_directory):
+        os.mkdir(output_directory)
+
+    # read in data to split for the folds
+    data_df = pd.read_csv(data_train_csv)
+
+    # place each fold in a separate directory
+    data_dir = os.path.dirname(data_train_csv)
+    kfold_training_stats = {}
+    for train_index, test_index, fold_num in generate_kfold_splits(data_train_csv, k_fold):
```

Comment:
maybe an option to save the train and test indices for each fold would be useful for reproducibility.

Reply:
Any particular preference on data format for storage?  json?  pickle?

Reply:
Decided to use json format to make it consistent with other saved information. Saved split indices saved to a data set called `kfold_split_indices.json` in the `output_directory`.  Format of data set looks like this
```
{
   "fold_1": {"test_index": [1, 3, 10,  ...], "training_index": [2, 4, 5,...]  },
   "fold_2": {"test_index": [...], "training_index": [...]},
   "fold_3": {"test_index": [...], "training_index": [...]}
}
```

added command line parameter to skip saving kfold indices. 
```
    parser.add_argument(
        '-skfsi',
        '--skip_save_k_fold_split_indices',
        action='store_true',
        default=False,
        help='disables saving indices generated to split training data set ' 
             'for the k-fold cross validation run, but if it is not needed '
             'turning it off can slightly increase the overall speed'
    )
```